lib/ExpensiMark.js

```diff
@@ -628,4 +612,25 @@ export default class ExpensiMark {
         }
         return replacedText;
     }
+
+    /**
+     * Handle whitespace for the content of blockquote if the text matches the regex or else just return the original text
+     *
+     * @param {RegExp} regex
+     * @param {String} textToCheck
+     * @param {Function} replacement
+     *
+     * @returns {String}
+     */
+    formatWhitespaceForQuote(regex, textToCheck, replacement) {
+        if (textToCheck.match(regex)) {
```

Comment:
I thought we agreed on using a regex-free implementation :sweat_smile: Do we really still need that extra regex?

Reply:
I think it's fine to use the same regex to test the input here. It'll be consistent and explicit to determine if the input text should be translated into a quote.

Reply:
I think we should avoid that for performance reasons. Regex is slow. Using regex for every line is an overkill.

Reply:
> Using regex for every line is an overkill.

Actually, we don't call regex matching on each row but on a bunch of rows that start with `&gt;`.

I think we don't need to concern about the regex performance issue here as the regex to match quote is moderate. What I feel about this lib is that it always prefers regex(conciseness provided by regex) to string modification.

Reply:
Oh actually we only call the regex once for each blockquote. So something like
```
Test
> hi
> there
Abc
> a
> b
> c
```

The regex call here is just 2, not 5 and not 7 :sweat_smile:

That's cool, I also like how the code is much simplified, will perform some more tests and get back to you